docs/langref/relay_expr.rst

```diff
@@ -0,0 +1,462 @@
+====================
+Expressions in Relay
+====================
+
+The Relay IR is a pure, expression-oriented language with distinct
+dataflow and control flow language fragments.
+Each dataflow fragment of a program (i.e., the portions of the
+program without recursive calls or branching) can be viewed as a
+traditional computation graph when writing and expressing transformations.
+
+The below sections describe the different expressions in Relay
+as well as give details of their semantics.
+
+Variables
+=========
+
+Inspired by LLVM, Relay explicitly distinguishes between local and
+global variables both in the AST and in the text format. In the text format,
+global and local variables are distinguished by prefixes, or sigils.
+Global variables are prefixed with :code:`@` and local variables with :code:`%`.
+
+This explicit distinction makes certain optimizations easier to implement.
+For example, inlining a global definition requires no analysis: simply
+substituting the definition suffices.
+
+Global Variable
+~~~~~~~~~~~~~~~~~~
+
+Global identifiers are prefixed by the :code:`@` sigil, such as ":code:`@global`".
+A global identifier always references a globally visible definition contained in the environment.
+Global identifiers must be unique.
+
+See :py:class:`~tvm.relay.expr.GlobalVar` for its implementation
+and documentation.
+
+Local Variable
+~~~~~~~~~~~~~~
+
+Local identifiers are prefixed by the :code:`%` sigil,
+such as ":code:`%local`". A local identifier always references
+a function argument or a variable bound in a :code:`let` expression,
+and will be scoped to the function where it appears or the :code:`let`
+expression where it is bound, respectively.
+
+In the below code segment, notice that :code:`%a` is defined twice. This is
+permitted, as in most functional languages; in the scope of the second
+:code:`let` expression, the name :code:`%a` is "shadowed," meaning all
+references to :code:`%a` in the inner scope refer to the later defintion, while
+references to :code:`%a` in the outer scope continue to refer to
+the first one.
+
+.. code-block:: python
+  def @f(%a) {
```

Comment:
What about

```
let %a = 1;
let %b = 2 * %a;  // %b = 2
let %a = %a + %a; // %a = 2. %a is shadowed
%a + %b           // has value 2 + 2 = 4
```

Reply:
Yeah, I was thinking about having comments in the code example. That would be good